Northstar.Client/mod/scripts/vscripts/ui/_menus.nut

```diff
@@ -178,6 +178,23 @@ void function UICodeCallback_ToggleInGameMenu()
 	}
 }
 
+
+void function TimeoutHandler()
+{
+	//Singleplayer "Continue" counts as finished loading and shouldn't be affected
+	printt( "TimeoutHandler has Initialized" )
+	EndSignal( uiGlobal.signalDummy, "LevelShutdown" ) // Cancels thread if map is unloaded
+
+	wait GetConVarFloat( "ns_max_load_time" )
+	if ( uiGlobal.isLoading == true ) //Disconnects client after 30 seconds of loading
+	{
+		printt( "TimeoutHandler disconnected because loading took too long" )
```

Comment:
```suggestion
		printt( format( "disconnecting after loading for %.1f seconds", maxLoadTime ) )
```

Reply:
LMK if you like the new print statement I put